src/ln/channelmanager.rs

```diff
@@ -501,11 +501,27 @@ impl ChannelManager {
 	/// Raises APIError::APIMisuseError when channel_value_satoshis > 2**24 or push_msat is
 	/// greater than channel_value_satoshis * 1k or channel_value_satoshis is < 1000.
 	pub fn create_channel(&self, their_network_key: PublicKey, channel_value_satoshis: u64, push_msat: u64, user_id: u64) -> Result<(), APIError> {
+		self.create_channel_with_id(their_network_key,rng::rand_u832(),channel_value_satoshis,push_msat,user_id)
+	}
+
+	/// Creates a new outbound channel to the given remote node and with the given value.
+	///
+	/// user_id will be provided back as user_channel_id in FundingGenerationReady and
+	/// FundingBroadcastSafe events to allow tracking of which events correspond with which
+	/// create_channel call. Note that user_channel_id defaults to 0 for inbound channels, so you
+	/// may wish to avoid using 0 for user_id here.
+	///
+	/// If successful, will generate a SendOpenChannel message event, so you should probably poll
+	/// PeerManager::process_events afterwards.
+	///
+	/// Raises APIError::APIMisuseError when channel_value_satoshis > 2**24 or push_msat is
+	/// greater than channel_value_satoshis * 1k or channel_value_satoshis is < 1000.
+	pub fn create_channel_with_id(&self, their_network_key: PublicKey, channel_id: [u8; 32], channel_value_satoshis: u64, push_msat: u64, user_id: u64) -> Result<(), APIError> {
```

Comment:
I really don't think this needs to be pub. In what case would a user ever care about specifying the temporary_channel_id explicitly?

Reply:
The only reason for this change was to make testing easier, but the change was reverted.